generated/osdep/meson.build

```diff
@@ -0,0 +1,68 @@
+xcrun = find_program('xcrun')
```

Comment:
maybe the whole file should be moved to a mac specific path, since it's mac only? generated/osdep/mac/meson.build

Reply:
This file is already mac-only. I can't really move it anywhere else because the `custom_target` can only create files in the subdirectory it is in. In this case, that would be `generated/osdep` which is needed for the headers to be correctly found.

Reply:
just for me and that i understood it correctly. since we are creating these files `osdep/macOS_swift.h`, `osdep/macOS_swift.swiftmodule` and `osdep/macOS_swift.o` under `osdep/` the `meson.build` file has to be in `osdep/` to be able to create those files in the same folder?

okay i think in the future (not within this PR) it would probably be wise to move those files to `osdep/mac` so we can also move the build file. otherwise it might get messy. if we add another `custom_target` in that folder, the file could contain totally unrelated functionality. do think that would be better?